pandas/core/indexes/datetimelike.py

```diff
@@ -587,11 +587,15 @@ def _concat_same_dtype(self, to_concat, name):
         if len({str(x.dtype) for x in to_concat}) != 1:
             raise ValueError('to_concat must have the same tz')
 
-        if not is_period_dtype(self):
+        new_data = type(self._values)._concat_same_type(to_concat).asi8
+
+        # GH 3232: If the concat result is evenly spaced, we can retain the
+        # original frequency
+        is_diff_evenly_spaced = len(np.unique(np.diff(new_data))) == 1
```

Comment:
I *think* we have something that does this in tseries.frequencies, can you see / use it?

Reply:
Checked in `tseries.frequencies` but didn't find anything. We have `is_monotonic` functions but it doesn't check for even spacing.

Reply:
you can use unique_1d here (should be faster), though not a big deal. worth moving this actually to tseries.frequencies as a named function?

Reply:
this looks a bit like `unique_deltas`

Reply:
Thanks @jbrockmendel. Exactly what I needed.